soh/soh/Enhancements/TimeDisplay/TimeDisplay.cpp

```diff
@@ -0,0 +1,255 @@
+#define COLOR_LIGHT_RED ImVec4(1.0f, 0.05f, 0, 1.0f)
+#define COLOR_LIGHT_BLUE ImVec4(0, 0.88f, 1.0f, 1.0f)
+#define COLOR_LIGHT_GREEN ImVec4(0.52f, 1.0f, 0.23f, 1.0f)
+#define COLOR_GREY ImVec4(0.78f, 0.78f, 0.78f, 1.0f)
+
+const static std::vector<std::pair<std::string, const char*>> digitList = {
+    { "DIGIT_0_TEXTURE", gCounterDigit0Tex }, { "DIGIT_1_TEXTURE", gCounterDigit1Tex },
+    { "DIGIT_2_TEXTURE", gCounterDigit2Tex }, { "DIGIT_3_TEXTURE", gCounterDigit3Tex },
+    { "DIGIT_4_TEXTURE", gCounterDigit4Tex }, { "DIGIT_5_TEXTURE", gCounterDigit5Tex },
+    { "DIGIT_6_TEXTURE", gCounterDigit6Tex }, { "DIGIT_7_TEXTURE", gCounterDigit7Tex },
+    { "DIGIT_8_TEXTURE", gCounterDigit8Tex }, { "DIGIT_9_TEXTURE", gCounterDigit9Tex },
+    { "COLON_TEXTURE", gCounterColonTex },
+};
+
+const std::vector<TimeObject> timeDisplayList = {
+    { DISPLAY_IN_GAME_TIMER, "Display Gameplay Timer", CVAR_ENHANCEMENT("TimeDisplay.Timers.InGameTimer") },
+    { DISPLAY_TIME_OF_DAY, "Display Time of Day", CVAR_ENHANCEMENT("TimeDisplay.Timers.TimeofDay") },
+    { DISPLAY_CONDITIONAL_TIMER, "Display Conditional Timer", CVAR_ENHANCEMENT("TimeDisplay.Timers.HotWater") },
+    { DISPLAY_NAVI_TIMER, "Display Navi Timer", CVAR_ENHANCEMENT("TimeDisplay.Timers.NaviTimer") }
+};
+
+static std::vector<TimeObject> activeTimers;
+
+std::string convertDayTime(uint32_t dayTime) {
+    uint32_t totalSeconds = 24 * 60 * 60;
+    uint32_t ss = static_cast<uint32_t>(static_cast<double>(dayTime) * (totalSeconds - 1) / 65535);
+    uint32_t hh = ss / 3600;
+    uint32_t mm = (ss % 3600) / 60;
+    return fmt::format("{:0>2}:{:0>2}", hh, mm);
+}
+
+std::string convertNaviTime(uint32_t value) {
+    uint32_t totalSeconds = value * 0.05;
+    uint32_t ss = totalSeconds % 60;
+    uint32_t mm = totalSeconds / 60;
+    return fmt::format("{:0>2}:{:0>2}", mm, ss);
+}
+
+std::string formatHotWaterDisplay(uint32_t value) {
+    uint32_t ss = value % 60;
+    uint32_t mm = value / 60;
+    return fmt::format("{:0>2}:{:0>2}", mm, ss);
+}
+
+std::string formatTimeDisplay(uint32_t value) {
+    uint32_t sec = value / 10;
+    uint32_t hh = sec / 3600;
+    uint32_t mm = (sec - hh * 3600) / 60;
+    uint32_t ss = sec - hh * 3600 - mm * 60;
+    uint32_t ds = value % 10;
+    return fmt::format("{}:{:0>2}:{:0>2}.{}", hh, mm, ss, ds);
+}
+
+static void TimeDisplayGetTimer(uint32_t timeID) {
+    timeDisplayTime = "";
+    textureDisplay = 0;
+    textColor = COLOR_WHITE;
+
+    Player* player = GET_PLAYER(gPlayState);
+    uint32_t timer1 = gSaveContext.timer1Value;
+
+    switch (timeID) {
+        case DISPLAY_IN_GAME_TIMER:
+            textureDisplay = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName("GAMEPLAY_TIMER");
+            timeDisplayTime = formatTimeDisplay(GAMEPLAYSTAT_TOTAL_TIME).c_str();
+            break;
+        case DISPLAY_TIME_OF_DAY:
+            if (gSaveContext.dayTime >= DAY_BEGINS && !(gSaveContext.dayTime >= NIGHT_BEGINS)) {
```

Comment:
One more thing, I think it would make more sense to have this be `gSaveContext.dayTime < NIGHT_BEGINS` rather than this !>=.

Reply:
doneskis